go/mcap/parse.go

```diff
@@ -99,32 +99,40 @@ func ParseChannel(buf []byte) (*Channel, error) {
 	}, nil
 }
 
-// ParseMessage parses a message record.
-func ParseMessage(buf []byte) (*Message, error) {
+// PopulateFrom populates the fields of a Message struct from the message data slice.
+func (m *Message) PopulateFrom(buf []byte) error {
 	channelID, offset, err := getUint16(buf, 0)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read channel ID: %w", err)
+		return fmt.Errorf("failed to read channel ID: %w", err)
 	}
 	sequence, offset, err := getUint32(buf, offset)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read sequence: %w", err)
+		return fmt.Errorf("failed to read sequence: %w", err)
 	}
 	logTime, offset, err := getUint64(buf, offset)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read record time: %w", err)
+		return fmt.Errorf("failed to read record time: %w", err)
 	}
 	publishTime, offset, err := getUint64(buf, offset)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read publish time: %w", err)
+		return fmt.Errorf("failed to read publish time: %w", err)
 	}
 	data := buf[offset:]
-	return &Message{
-		ChannelID:   channelID,
-		Sequence:    sequence,
-		LogTime:     logTime,
-		PublishTime: publishTime,
-		Data:        data,
-	}, nil
+	m.ChannelID = channelID
+	m.Sequence = sequence
+	m.LogTime = logTime
+	m.PublishTime = publishTime
+	m.Data = data
```

Comment:
if msg.Data was oversized, we drop the excess capacity here, which will cause more allocations as big and small messages alternate. Ideally I think we want the capacity of the msg.Data array to be monotonically increasing across calls. I'm not sure if our benchmarks surface this situation.